Sources/Time.swift

```diff
@@ -0,0 +1,64 @@
+#if os(Linux)
+    import Glibc
+#else
+    import Darwin.C
+    var mtid = mach_timebase_info_data_t()
+#endif
+
+/// Milliseconds from January 1, 1970.
+public func now() -> Int {
+    #if os(Linux)
+        var ts = timespec()
+        clock_gettime(CLOCK_MONOTONIC, &ts)
+        return Int(ts.tv_sec) * 1000 + Int(ts.tv_nsec) / 1000000
+    #else
+        if mtid.denom == 0 {
+            mach_timebase_info(&mtid)
+        }
+        let ticks = mach_absolute_time()
+        return Int(ticks * UInt64(mtid.numer) / UInt64(mtid.denom) / 1000000)
+    #endif
+}
+
+public extension Int {
+    /// Interval of `self` from now.
+    func fromNow() -> Int {
+        return now() + self
+    }
+}
+
+public protocol TimeRepresentable {
+    var millisecond: Int { get }
+}
+
+extension TimeRepresentable {
+    public var milliseconds: Int {
+        return 1
+    }
+    public var second: Int {
+        return millisecond * 1000
+    }
+    public var seconds: Int {
+        return 1000
+    }
+    public var minute: Int {
+        return millisecond * 60000
+    }
+    public var minutes: Int {
+        return 60000
+    }
+    public var hour: Int {
+        return millisecond * 3600000
+    }
+    public var hours: Int {
+        return 3600000
+    }
+}
+
+extension Int: TimeRepresentable {
+    public var millisecond: Int {
+        return self
+    }
+}
+
+public let never = -1
```

Comment:
I think it’s better to be:

```swift
extension Int {
    public static var never: Int {
        return -1
    }
}
```

And so for some `func some(duration: Int)` it will be possible to write `some(.never)`.